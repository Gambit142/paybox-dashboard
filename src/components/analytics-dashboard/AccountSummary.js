import styles from './account-summary.module.css';
import accountDetails from '../../data/accountDetails';

const AccountSummary = () => (
  <section className={`flex flex-column ${styles.accountSummaryContainer}`}>
    <div className={`${styles['balance-container']} flex flex-column secondary-background`}>
      <div className={`${styles.image} ${styles.image1}`} />
      <div className={`${styles.image} ${styles.image2}`} />
      <div className={`${styles.image} ${styles.image3}`} />
      <div className={`${styles.image} ${styles.image4}`} />
      <div className={`nunito-font-400 flex flex-column fs-14 ${styles.walletBalText} ${styles.overlay}`}>
        <h4>Wallet Balance</h4>
        <div className={`sora-font-600 white fs-20 flex font-numeric ${styles.walletBalAmt}`}>
          <span>â‚¦50,000</span>
          <img src="/assets/images/eye.svg" alt="eye icon" />
        </div>
        <div className={`${styles.walletBtnDiv} flex`}>
          <button type="button" className={`nunito-font-600 fs-14 white ${styles.btn1}`}>
            Add Fund
          </button>
          <button type="button" className={`nunito-font-600 fs-14 white ${styles.btn2}`}>
            Withdraw
          </button>
        </div>
      </div>
    </div>
    {accountDetails.map((detail) => (
      <article className={`${styles.accountDetailsContainer}`} key={detail.detailType}>
        <div className={`${styles.accountDetails} flex flex-column`}>
          <span className={`light-blue sora-font-400 fs-14 font-numeric ${styles.detailInfo} ${styles.detailType}`}>{detail.detailType}</span>
          <span className={`dark-blue font-numeric sora-font-600 fs-20 ${styles.detailInfo} ${styles.detailValue}`}>{detail.detailAmount}</span>
        </div>
        <div className="flex">
          <button type="button" className={`dark-blue font-numeric nunito-font-400 fs-12 ${styles.detailType} ${styles.detailInfo} ${styles.detailsBtn}`}>
            View Details
          </button>
          <div className={styles.rightIconDiv}>
            <img src="./assets/images/group-right-icon.svg" alt="group-right-icon" />
          </div>
        </div>
      </article>
    ))}
  </section>
);

AccountSummary.displayName = 'AccountSummary';

export default AccountSummary;
