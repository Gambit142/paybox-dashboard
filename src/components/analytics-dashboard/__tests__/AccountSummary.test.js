import { render, screen, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AccountSummary from '../AccountSummary';
import '@testing-library/jest-dom';

jest.mock('../../../data/accountDetails', () => [

  {
    detailType: 'Total savings balance',
    detailAmount: '₦194,000',
  },
  {
    detailType: 'Total customers',
    detailAmount: '₦919',
  },
  {
    detailType: 'Total POS transactions',
    detailAmount: '0',
  },
]);

describe('AccountSummary', () => {
  test('AccountSummary renders correctly', () => {
    const tree = renderer.create(<AccountSummary />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('AccountSummary renders wallet balance', async () => {
    render(<AccountSummary />);
    await waitFor(() => {
      const walletBalanceElement = screen.getByText('Wallet Balance');
      expect(walletBalanceElement).toBeInTheDocument();
    });
  });

  test('AccountSummary renders account details', async () => {
    render(<AccountSummary />);
    await waitFor(() => {
      // const totalSavings = screen.getByText('Total savings balance');
      // const totalCustomers = screen.getByText('Total customers');
      // const totalPOSTransactions = screen.getByText('Total POS transactions');
      expect(screen.getByText('Total savings balance')).toBeInTheDocument();
      expect(screen.getByText('Total customers')).toBeInTheDocument();
      expect(screen.getByText('Total POS transactions')).toBeInTheDocument();
    });
  });
});
