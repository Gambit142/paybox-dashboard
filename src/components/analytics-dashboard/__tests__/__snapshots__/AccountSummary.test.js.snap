// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountSummary AccountSummary renders correctly 1`] = `
<section
  className="flex flex-column accountSummaryContainer"
>
  <div
    className="balance-container flex flex-column secondary-background"
  >
    <div
      className="nunito-font-400 flex flex-column fs-14 walletBalText overlay"
    >
      <h4>
        Wallet Balance
      </h4>
      <div
        className="sora-font-600 white fs-20 flex font-numeric walletBalAmt"
      >
        <span>
          ₦50,000
        </span>
        <img
          alt="eye icon"
          src="/assets/images/eye.svg"
        />
      </div>
      <div
        className="walletBtnDiv flex"
      >
        <button
          className="nunito-font-600 fs-14 white btn1"
          type="button"
        >
          Add Fund
        </button>
        <button
          className="nunito-font-600 fs-14 white btn2"
          type="button"
        >
          Withdraw
        </button>
      </div>
    </div>
  </div>
  <article
    className="accountDetailsContainer"
  >
    <div
      className="accountDetails flex flex-column"
    >
      <span
        className="light-blue sora-font-400 fs-14 font-numeric detailInfo detailType"
      >
        Total savings balance
      </span>
      <span
        className="dark-blue font-numeric sora-font-600 fs-20 detailInfo detailValue"
      >
        ₦194,000
      </span>
    </div>
    <div
      className="flex"
    >
      <button
        className="dark-blue font-numeric nunito-font-400 fs-12 detailType detailInfo detailsBtn"
        type="button"
      >
        View Details
      </button>
      <div
        className="rightIconDiv"
      >
        <img
          alt="group-right-icon"
          src="./assets/images/group-right-icon.svg"
        />
      </div>
    </div>
  </article>
  <article
    className="accountDetailsContainer"
  >
    <div
      className="accountDetails flex flex-column"
    >
      <span
        className="light-blue sora-font-400 fs-14 font-numeric detailInfo detailType"
      >
        Total customers
      </span>
      <span
        className="dark-blue font-numeric sora-font-600 fs-20 detailInfo detailValue"
      >
        ₦919
      </span>
    </div>
    <div
      className="flex"
    >
      <button
        className="dark-blue font-numeric nunito-font-400 fs-12 detailType detailInfo detailsBtn"
        type="button"
      >
        View Details
      </button>
      <div
        className="rightIconDiv"
      >
        <img
          alt="group-right-icon"
          src="./assets/images/group-right-icon.svg"
        />
      </div>
    </div>
  </article>
  <article
    className="accountDetailsContainer"
  >
    <div
      className="accountDetails flex flex-column"
    >
      <span
        className="light-blue sora-font-400 fs-14 font-numeric detailInfo detailType"
      >
        Total POS transactions
      </span>
      <span
        className="dark-blue font-numeric sora-font-600 fs-20 detailInfo detailValue"
      >
        0
      </span>
    </div>
    <div
      className="flex"
    >
      <button
        className="dark-blue font-numeric nunito-font-400 fs-12 detailType detailInfo detailsBtn"
        type="button"
      >
        View Details
      </button>
      <div
        className="rightIconDiv"
      >
        <img
          alt="group-right-icon"
          src="./assets/images/group-right-icon.svg"
        />
      </div>
    </div>
  </article>
</section>
`;
